C-sharp/demo/src/common/csConsoleApplicationJournalREST/Program.cs

```diff
@@ -133,7 +140,7 @@ static void Journal(MenuItem item)
                     {
                         readCount = (readCount / 2) + 1;
                     }
-                } while (pointer > 0);
+                } while (pointer >= 0 & readCount > 50);
```

Comment:
Logical _or_?

Reply:
Aside from this, could we make this code cleaner? It looks quite complicated to me. Maybe introduce a method, use constants, etc.

Reply:
Good catch also here.
Changed the code to give a clearer meaning.